docs/code-analysis/language-server-protocol/rosie-language-server.md

```diff
@@ -0,0 +1,178 @@
+                "codiga.api.token": {
+                    "type": "string",
+                    "default": "<API-TOKEN>",
+                    "description": "Your Codiga API Token",
+                    "minLength": 32,
+                    "maxLength": 50,
+                    "scope": "application"
+                }
+            }
+        }
+    ]
+```
+
+### User Fingerprint
+
+The fingerprint is used for telemetry to improve the user experience. It is anonymized, and does not disclose the user identity.
+It is in place to know what rulesets are being used, and if they are used by the same user without sending the exact user identifier.
+
+Ideally, this value is generated when an IDE/text editor plugin is installed, and that is also stored and kept between subsequent
+client application launches.
+
+The language server provides logic to generate this value, but for now it is re-generated on each new server launch.
+
+You can implement the generation of the fingerprint on client side, and if you do so, you must pass it in, in a command line argument
+called `fingerprint`, for the server, like for example:
+
+```
+fingerprint=<generated value>
+```
+
+Example from VS Code extension:
+
+```typescript
+const serverOptions: ServerOptions = {
+    run: { ..., args: [`fingerprint=${getUserFingerprint()}`] },
+    debug: { ..., args: [`fingerprint=${getUserFingerprint()}`] }
+  };
+```
+
+The VS Code implementation of the fingerprint generation can be found at [codiga/vscode-plugin: configurationUtils.ts](https://github.com/codiga/vscode-plugin/blob/main/src/utils/configurationUtils.ts). 
+
+
+## Configure the Language Client
+
+### Supported file types
+
+When integrating with a client application, make sure to enable languages/file types/mime types that are supported by Rosie,
+so that they actually get analyzed by the language server. Please note that this might not be necessary in all client implementations. 
+
+Example from the Codiga VS Code extension:
+
+```typescript
+const clientOptions: LanguageClientOptions = {
+    documentSelector: [
+      { scheme: 'file', language: 'javascript' },
+      { scheme: 'file', language: 'javascriptreact' },
+      { scheme: 'file', language: 'typescript' },
+      { scheme: 'file', language: 'typescriptreact' },
+      { scheme: 'file', language: 'python' }
+    ]
+};
+```
+
+**Notes:**
+- the language identifiers, mime types may differ in different client applications.
+- list of supported languages: JavaScript, JavaScript+React, TypeScript, TypeScript+React, Python (including Jupyter notebooks)
+
+See the related [DocumentFilter](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#documentFilter) section in the LSP Specification.
+
+### Communication kind
+
+Depending on the modes (`--node-ipc`, `--stdio`, etc.) the client application supports, you will have to use either one
+of them passed in as a command line argument for the server.
```

Comment:
Could we add some text/links regarding the `TransportKind`? 

It was unclear that we want to pass a number here

```
enum TransportKind {
    stdio = 0,
    ipc = 1,
    pipe = 2,
    socket = 3
}
```